core/src/main/scala/cats/Selective.scala

```diff
@@ -0,0 +1,78 @@
+package cats
+
+import simulacrum.{noop, typeclass}
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("Could not find an instance of Selective for ${F}")
+@typeclass trait Selective[F[_]] extends Applicative[F] {
+  def select[A, B](fab: F[Either[A, B]])(ff: => F[A => B]): F[B]
+
+  def branch[A, B, C](fab: F[Either[A, B]])(fl: => F[A => C])(fr: => F[B => C]): F[C] = {
+    val innerLhs: F[Either[A, Either[B, C]]] = map(fab)(_.map(Left(_)))
+    def innerRhs: F[A => Either[B, C]] = map(fl)(_.andThen(Right(_)))
+    val lhs = select(innerLhs)(innerRhs)
+    select(lhs)(fr)
+  }
+
+  @noop
+  def ifS[A](fCond: F[Boolean])(fTrue: => F[A])(fFalse: => F[A]): F[A] = {
```

Comment:
this is the exact same signature as ifM. Why can't we use the same name?

This is going to be really confusing. I would rather just name this method ifM.

Reply:
If this were a clean slate, we'd pull `ifM` down and call it `ifS`.  `ifM` is equivalent to `ifS` where it presently exists, but if we pull it down, some implementations will behave like `ifA`, and we'd be unexpectedly changing the behavior of an established name.

I would like to call it `ifS` and deprecate `ifM`, but I'm not sure how bad the warnings would be.